src/vm.rs

```diff
@@ -1393,6 +1426,52 @@ impl<T: GuestStagePageTable> HostVm<T, VmStateInitializing> {
         }
     }
 
+    /// Adds the IMSIC pages for `cpu` to the host. The first page in `pages` is set as the
+    /// host's interrupt file for `cpu` while the remaining pages are added as guest interrupt
+    /// files for the host to assign.
+    pub fn add_imsic_pages<I>(&mut self, cpu: CpuId, pages: I)
+    where
+        I: ExactSizeIterator<Item = ImsicGuestPage<ConvertedClean>>,
+    {
+        // We assigned an IMSIC geometry and vCPU IMSIC locations in `from_hyp_mem()`.
+        let location = self
+            .inner
+            .get_vcpu_imsic_location(cpu.raw() as u64)
+            .unwrap();
+        let to_addr = self
+            .inner
+            .vm_pages
+            .imsic_geometry()
+            .and_then(|g| g.location_to_addr(location))
+            .unwrap();
+        // Unwrap ok since we've donated sufficient PT pages to map the entire address space up
+        // front.
+        let mapper = self
+            .inner
+            .vm_pages
+            .map_pages(to_addr, pages.len() as u64, VmRegionType::Imsic)
+            .unwrap();
+        let page_tracker = self.inner.vm_pages.page_tracker();
+        for (i, (page, vm_addr)) in pages.zip(to_addr.iter_from()).enumerate() {
+            if i == 0 {
+                // Set the first page as the vCPU's supervisor-level interrupt file.
+                self.inner
+                    .bind_vcpu(cpu.raw() as u64, page.location().file())
+                    .unwrap();
+            }
+
+            // Map in the remaining guest interrupt file pages as guest files for the host VM.
+            //
```

Comment:
Nit: Remove blank comment line unless spacing before `// TODO` is the norm.

Reply:
I've put the blank line to logically separate the TODO from the rest of the comment.